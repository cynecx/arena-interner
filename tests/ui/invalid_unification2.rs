use arena_interner::{make_interner, Interned};

fn unify2<'a, 'b, 'c>(a: Interned<'a, 'c, str>, b: Interned<'b, 'c, str>) {
    assert_eq!(a, b);
}

fn main() {
    make_interner!(interner1: str);
    make_interner!(interner2: str);

    let a = interner1.intern("hello");
    let b = interner2.intern("hello");

    unify2(a, b);
}
