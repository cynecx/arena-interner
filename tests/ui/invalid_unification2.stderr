error[E0597]: `tag` does not live long enough
  --> tests/ui/invalid_unification2.rs:9:5
   |
9  |     make_interner!(interner2: str);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ borrowed value does not live long enough
...
15 | }
   | -
   | |
   | `tag` dropped here while still borrowed
   | borrow might be used here, when `_guard` is dropped and runs the `Drop` code for type `main::make_guard`
   |
   = note: values in a scope are dropped in the opposite order they are defined
   = note: this error originates in the macro `$crate::internal::generativity::make_guard` which comes from the expansion of the macro `make_interner` (in Nightly builds, run with -Z macro-backtrace for more info)
